client/components/Checkout.js

```diff
@@ -0,0 +1,447 @@
+/* eslint-disable indent */
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import {
+  fetchOrder as fetchOrderThunk,
+  fetchOrderItems,
+  updateOrderThunk,
+  updateUser,
+} from '../store';
+import { isLoggedIn } from './helperFunctions';
+
+class Checkout extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      shippingAddress: '',
+      shippingCity: '',
+      shippingState: '',
+      shippingZip: '',
+      billingFirstName: '',
+      billingLastName: '',
+      billingAddress: '',
+      billingCity: '',
+      billingState: '',
+      billingZip: '',
+      ccNumber: '',
+      ccv: '',
+      ccExpDateMonth: 1,
+      ccExpDateYear: 2019,
+      sameAddress: false,
+      submitted: false,
+    };
+  }
+
+  componentDidMount = () => {
+    const { fetchOrder, user, order } = this.props;
+    fetchOrder(user.id);
+    fetchOrderItems(order.id);
+  };
+
+  handleChange = ev => {
+    ev.preventDefault();
+
+    const {
+      firstName,
+      lastName,
+      shippingAddress,
+      shippingCity,
+      shippingState,
+      shippingZip,
+      sameAddress,
+    } = this.state;
+
+    this.setState({ [`${ev.target.name}`]: ev.target.value });
+
+    if (sameAddress) {
```

Comment:
I believe I generally have seen you enter your billing address first and if it is the same then the billing info would be copied to the shipping info.  You did the reverse.  Obviously it still works this way.  Perhaps we can discuss this at our meeting to figure this out.